qemu/tests/thin_provision_check_mode.py

```diff
@@ -0,0 +1,98 @@
+import os
+import logging
+
+from avocado.utils import genio
+from avocado.utils import path as utils_path
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import error_context
+
+from virttest.utils_misc import get_linux_drive_path
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu provisioning mode checking test:
+    1) load scsi_debug module with lbpu=1 / lbpu=0
+    2) boot guest with scsi_debug emulated disk as extra data disk
+    3) get provisioning mode of data disk in guest
+    4) check provisioning mode
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_host_scsi_disk():
+        """
+        Get latest scsi disk which emulated by scsi_debug module.
+        """
+        scsi_disk_info = process.system_output("lsscsi").decode().splitlines()
+        scsi_debug = [_ for _ in scsi_disk_info if 'scsi_debug' in _][-1]
+        scsi_debug = scsi_debug.split()
+        host_id = scsi_debug[0][1:-1]
+        device_name = scsi_debug[-1]
+        return host_id, device_name
```

Comment:
this command has executed before. and i find better method, please check code

Reply:
You wanna execute it before booting up qemu, so you could create a sg device before     `params["start_vm"] = "yes"` since the the qemu will not boot up before` params["start_vm"] = "yes"`

Such as:
```
        - unmap:
           create_command = "modprobe -r scsi_debug; modprobe scsi_debug dev_size_mb=${disk_size} lbpu=1 lbpws=1 lbprz=0"
        - writesame:
           create_command = "modprobe -r scsi_debug; modprobe scsi_debug dev_size_mb=${disk_size} lbpu=0 lbpws=1 lbprz=0"
```

```
    def get_host_scsi_disk():
        """
        Get  scsi disk which emulated by scsi_debug module.
        """
        disks_old = process.getoutput("ls -1d /dev/sd*", shell=True).split()
        process.system_output(params["create_command"], timeout=300,
                              shell=True, verbose=False)
        disks_new = process.getoutput("ls -1d /dev/sd*", shell=True).split()
        new_disk = list(set(disks_new) ^ set(disks_old))
        if len(new_disk) != 1:
            test.error('No found the emulated scsi disk.')
        cmd = "lsblk -S -n -p|grep %s" % new_disk[0]
        return process.getoutput(cmd).strip().split()[1], new_disk[0]
```